### PR TITLE
Release Candidate - azblob v0.4.0

### PR DESCRIPTION
**Pull request by mohsha-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by mohsha-msft** [comment]
@siminsavani-msft , we'll use `dev-azblob` as target branch for our release.

File: sdk/storage/azblob/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Features Added
 
 ### Breaking Changes
```
Comment (seankane-msft) [inline sdk/storage/azblob/CHANGELOG.md:7]: No further changes made in this API?

File: sdk/storage/azblob/README.md [hunk 2]
```diff
@@ -205,91 +205,91 @@ Three different clients are provided to interact with the various components of
...
-	handle(err)
-
-	// Delete the container we created earlier.
-	_, err = container.Delete(ctx, nil)
-	handle(err)
+    // Use your storage account's name and key to create a credential object, used to access your account.
+// You can obtain these details from the Azure Portal.
+accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+if !ok {
+handle(errors.New("AZURE_STORAGE_ACCOUNT_NAME could not be found"))
```
Comment (seankane-msft) [inline sdk/storage/azblob/README.md:212]: can we indent on these? Unless markdown does this automatically

File: sdk/storage/azblob/autorest.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+openapi-type: "data-plane"
+verbose: true
+security: AzureKey
+module-version: "0.3.0"
+modelerfour:
+  group-parameters: false
+  seal-single-value-enum-by-default: true
+  lenient-model-deduplication: true
+export-clients: false
+use: "@autorest/go@4.0.0-preview.36"
```
Comment (seankane-msft) [inline sdk/storage/azblob/autorest.md:32]: Any reason we're not using the latest version (preview.38) ?

File: sdk/storage/azblob/doc.go [hunk 4]
```diff
@@ -105,13 +105,13 @@ There are three different clients provided to interact with the various componen
     * Get and set account settings.
     * Query, create, and delete containers within the account.
 
-2. **`ContainerClient`**
+2. **`containerClient`**
```
Comment (seankane-msft) [inline sdk/storage/azblob/doc.go:108]: these should be capitalized otherwise they're not exported.

**Pull request closed without merging**